src/utils/rest-api.js

```diff
@@ -54,16 +54,35 @@ export function connectNotificationsWsUpdateConfig() {
     return reconnectingWebSocket;
 }
 
-function backendFetch(url, init) {
+function handleResponse(response, isJson) {
+    if (response.ok) {
+        return isJson ? response.json() : response;
+    } else {
+        return response.text().then((text) => {
+            return Promise.reject({
+                message: text ? text : response.statusText,
+                status: response.status,
+                statusText: response.statusText,
+            });
+        });
+    }
+}
+
+function backendFetch(url, init, isJson, withAuth = true) {
```

Comment:
I suggest inverting order between 'init' and 'isJson' (that I would call 'expectsJson', did I already tell it ? ;-) ) as the shorter boolean and yet mandatory literal would provoke less reindenting of the numerous calls downward in the diff. 'init' could then even be declared with the _default_ value of undefined bringing less noise in the calls.